uiautomationutilities/build.gradle

```diff
@@ -44,14 +53,78 @@ android {
 
 }
 
+
+task sourcesJar(type: Jar) {
+    from android.sourceSets.main.java.srcDirs
+}
+
+tasks.whenTaskAdded { task ->
+    if (task.name == 'assembleDebug' || task.name == 'assembleRelease') {
+        task.dependsOn 'sourcesJar'
+    }
+}
+
+project.afterEvaluate{
+    publishing {
+        publications {
+            aar(MavenPublication) {
+                artifact sourcesJar
+                groupId 'com.microsoft.identity'
+                artifactId = 'uiautomationutilities'
+                version = getAppVersionName()
+
+                pom.withXml {
+
+                    // Dependencies
+                    def dependenciesNode = asNode().appendNode('dependencies')
+
+                    //Iterate over the implementation dependencies (we don't want the test ones), adding a <dependency> node for each
+                    configurations.implementation.allDependencies.each {
+                        if (it.group != null && it.name != null) {
+                            def dependencyNode = dependenciesNode.appendNode('dependency')
+                            if(it.group == 's'){
+                                dependencyNode.appendNode('groupId', groupId)
+                            }else{
+                                dependencyNode.appendNode('groupId', it.group)
+                            }
```

Comment:
should we supply this as a function?

Reply:
Going to see if I can resolve this in a different way.